_inc/client/components/data/query-site/index.jsx

```diff
@@ -27,8 +26,8 @@ export const QuerySite = React.createClass( {
 export default connect(
 	( state ) => {
 		return {
-			isFetchingSiteData: isFetchingSiteData( state ),
-			siteData: fetchSiteData( state )
+			isFetchingSiteData: isFetchingSiteData( state.jetpack.siteData ),
```

Comment:
This should stay like 
```
isFetchingSiteData: isFetchingSiteData( state )
```

Path resolution inside the `state` should be accomplished by `isFetchingSiteData` itself.